src/iconv.cr

```diff
@@ -0,0 +1,39 @@
+@[Link("iconv")]
+lib LibIconv
+  type Iconv = Void*
+
+  fun open = iconv_open(tocode : LibC::Char*, fromcode : LibC::Char*) : Iconv
+  fun iconv(cd : Iconv, inbuf : LibC::Char**, inbytesleft : LibC::SizeT*, outbuf : LibC::Char**, outbytesleft : LibC::SizeT*) : LibC::SizeT
+  fun close = iconv_close(cd : Iconv) : LibC::Int
+end
+
+# :nodoc:
+struct Iconv
+  def initialize(from : String, to : String)
+    @iconv = LibIconv.open(to, from)
+    if @iconv.address == -1
+      raise ArgumentError.new("invalid encoding")
```

Comment:
Let's include the invalid encoding in the error message, so it's easier to spot (be able to just grep for it in the source).